src/render/extract.rs

```diff
@@ -1,25 +1,34 @@
-use bevy::{prelude::*, render::Extract};
-
-use crate::prelude::Grass;
-
 use super::cache::{EntityCache, GrassCache};
+use crate::prelude::Grass;
+use bevy::{prelude::*, render::Extract, utils::HashSet};
 
+/// Extracts the grass data into the render world.
+///
+/// The extraction only happens on change or creation of the entity,
+/// so it normally doesn't come at a high performance cost.
+///
+/// Note:
+/// 1) Currently, the grass data extracted in the render world doesn't get freed when the grass entity is deleted.
```

Comment:
Are you sure you want to leave it like this? Imagine a player entering a house -> grass is deleted -> player exists -> grass is added again -> etc.

Reply:
No. There are actually other problems with memory too. 
However I feel like the implementation wouldn't be trivial to solve all these problems.
I'll create an issue for that.
For now I'm fine with it